i2c-communication/robot.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+"""
+    This is a sample program demonstrating how to communicate to a light controller from the robot
+    code using the roboRIO's I2C port.
+"""
+
+import wpilib
+
+import string
+
+
+class MyRobot(wpilib.TimedRobot):
+    PORT = wpilib.I2C.Port.kOnboard
+    DEVICE_ADDRESS = 4
+
+    def robotInit(self):
+        """Robot initialization function"""
+
+        self.arduino = wpilib.I2C(self.PORT, self.DEVICE_ADDRESS)
+
+    def writeString(self, input: string):
+        # Creates a char array from the input string
+        chars = bytearray(input, "ascii")
+
+        # Writes bytes over I2C
+        self.arduino.writeBulk(chars)
```

Comment:
input is a python function, so let's call this variable something else, like... `s`? 

You can do `chars.encode('utf-8')` for this. `bytearray` is really for things that need to be modified.

Reply:
Good to know, thanks!